### PR TITLE
kafka_robustness

### PR DESCRIPTION
**Pull request by JKL98ISR**

### CODE REVIEW

File: backend/deepchecks_monitoring/logic/data_ingestion.py [hunk 1]
```diff
@@ -347,84 +347,100 @@ def __init__(
...
+                messages = failed_messages
+                retry_count += 1
+
+                if retry_count < max_retries:
+                    await asyncio.sleep(retry_count)
+            else:
+                break
+
+        if failed_messages:
+            self.logger.error(f"Failed to send {len(failed_messages)} messages after {max_retries} retries.")
```
Comment (shayts7) [inline backend/deepchecks_monitoring/logic/data_ingestion.py:377]: better to say "out of... (number of total messages)"

File: backend/deepchecks_monitoring/logic/data_ingestion.py [hunk 2]
```diff
@@ -347,84 +347,100 @@ def __init__(
         self.logger = logger or configure_logger(name="data-ingestion")
         self.use_kafka = self.resources_provider.kafka_settings.kafka_host is not None
 
+    async def _send_with_retry(self, producer: AIOKafkaProducer, topic_name: str, messages: list, max_retries=3):
```
Comment (shayts7) [inline backend/deepchecks_monitoring/logic/data_ingestion.py:350]: messages: list[tuple]

File: backend/deepchecks_monitoring/resources.py [hunk 3]
```diff
@@ -393,6 +396,13 @@ def shutdown_parallel_check_executors_pool(self):
         import ray  # noqa
         ray.shutdown()
 
+    @tenacity.retry(
```
Comment (shayts7) [inline backend/deepchecks_monitoring/resources.py:399]: why do we need the retry here?

**Review by shayts7 — APPROVED** [review]
See comments inside

File: backend/deepchecks_monitoring/resources.py (hunk 3 above)
Comment (JKL98ISR) [reply backend/deepchecks_monitoring/resources.py:399]: If it failes due to connection error or such

**Pull request merged**